### DISCUSSION TITLE
Monthly export

### DISCUSSION THREAD

**Discussion by DTB-K** [davetapley/helium-tax#80] [Ideas]
I'd find it useful (and I wondered if others might too) to be able to export for a specific month & year combo.  Even just a "Last month" option would be useful if nothing else.

Thanks

David

**Comment by davetapley** [comment]
I'd consider this if there were enough requests, but it's not that hard to just manually discard (or pivot table) the relevant rows in a spreadsheet application?

**Reply by michael2112** [reply]
That is a workaround that was good up to the point the general Helium API limitation issues (and stability?) kicked in and the process would stop.  I would download the whole year, then find the entries in the csv that were past the last date/time in my existing spreadsheet (which also contained prior year transactions as well so that I could see the continuing trend).  But, if making the download smaller & quicker by requesting data a month at a time would help the process complete, it would be useful.

**Reply by davetapley** [reply]
> if making the download smaller & quicker by requesting data a month at a time would help the process complete,

Yeah intuitively this seems like a good solution, but in my experience the rate limiting is _very inconsistent_; some times you'd barely get a day's worth and sometimes it would do the whole year just fine. Probably not worth the added complexity.

**Comment by michael2112** [comment]
I second the motion.  

My situation is such that my taxes are done on a fiscal year end (3/31 in my case), not a calendar year. Plus, if the earnings throughout the year effect a person's (or business's) income enough, they could also need the information to file quarterly estimated tax payments.

I am not enough of a programmer to be able to tell if making the output download smaller would help performance on your system or against whatever API you may be using, but maybe that would be a benefit too. Thank you...!
